src/borg/archive.py

```diff
@@ -774,6 +774,7 @@ def write_part_file(self, item, from_chunk, number):
         length = len(item.chunks)
         # the item should only have the *additional* chunks we processed after the last partial item:
         item.chunks = item.chunks[from_chunk:]
+        item.size = sum(chunk.size for chunk in item.chunks)
```

Comment:
this "sum of chunk sizes" function is duplicated at multiple places.
could be moved to helpers.py or item.pyx (can it be sped up with cython?).